ships/0024-surfacing-errors-to-buildrun.md

```diff
@@ -117,11 +117,56 @@ As a third-party application developer that uses shipwright under the hood, I wa
 
 ### Implementation Notes
 
-`tbd in the next phase of the proposal`
+
+This ship proposes two changes in the API that are implemented in `buildrun_types.go` and thus propagated in the CRD. First, the BuildRun's status gets the new field `errorDetails` that embeds the `failedAt` field:
+
+```go
+type FailureDetails struct {
+    Reason   string    `json:"reason,omitempty"`
+    Message  string    `json:"message,omitempty"`
+    Location *FailedAt `json:"location,omitempty"`
+}
+``` 
+
+Second, we keep the `failedAt` field to prevent breaking user applications. However, it should be deprecated and removed in the future due to its redundancy. Therefore, we propose adding documents that state the deprecation status and adding the news to the release notes.
+To ease deprecation and future removal, we propose refactoring `UpdateBuildRunUsingTaskRunCondition` by extracting the logic of finding the failed pod and container in a BuildRun/TaskRun.
+The new file `pkg/reconciler/buildrun/resources/failures.go` includes the extraced logic and other failure related functions.
+
+A TaskRun can emit failures using two results:
+1. shp-error-reason
+2. shp-error-message
```

Comment:
We should ensure that this is documented as a condition of release readiness.

Reply:
I am wondering whether this is possible for all build strategies. In case of the `git` command, we control the binary and can easily provide information. For example, the image in the _build and push_ step in the kaniko strategy would be really though to customize to provide those two values. In my opinion, they are useful hints and should be provided if possible. However, we are not able to introduce them to all possible scenarios since the binaries/images can be out of control.